Lib/MitreId.php

```diff
@@ -246,5 +257,73 @@ public static function insertNewEntitlements($mitreId, $user_id, $current_entitl
       }
       $mitreId->query('INSERT INTO user_edu_person_entitlement (user_id, edu_person_entitlement) VALUES ' . substr($insertEntitlementsParam, 0, -1));
     }
+    if($mitreId->rciamExternalEntitlements) {
+      // Import the ones from third parties
+      MitreId::insertRciamSyncVomsEntitlements($mitreId, $user_id);
+    }
+  }
+
+  /**
+   * @param $mitreId
+   * @param $mitre_user_id
+   */
+  public static function insertRciamSyncVomsEntitlements($mitreId, $mitre_user_id)
+  {
+    // Link to my table
+    $mdl_name = Inflector::camelize(MitreIdProvisionerRciamSyncVomsCfg::TableName);
+    // XXX Since this table uses the useDbConfig = "default" i can not use
+    // CAKEPHP's embeded PDO because it will always add the prefix cm_
+    $RciamModel = ClassRegistry::init($mdl_name);
+    // Get configuration
+    $vo_roles = !empty($mitreId->voRoles) ? explode(",", $mitreId->voRoles) : array();
+
+    $entitlements = array();
+    // Fetch users Certificates
+    if(empty($mitreId->userProfile['Cert'])) {
```

Comment:
maybe put that if condition at the beginning of the function

Reply:
Done